packages/react/src/components/form/form.tsx

```diff
@@ -14,42 +14,134 @@ import {
+ */
+export const Form = ({
   className,
   onSuccess,
   factors,
   text,
-  middleware
-}) => {
+  middleware,
+  children,
+}: Props) => {
   const flowState = useFlowState({ onSuccess });
   const { showBanner } = useConfiguration();
   const { lastHandle } = useLastHandle();
+  const submitPayloadRef = React.useRef<PayloadOptions>({
+    handleType: undefined,
+    handleValue: undefined,
+    flag: undefined,
+  });
+  const [selectedFactor, setSelectedFactor] = React.useState<
+    Factor | undefined
+  >();
+  const { status } = flowState;
+
+  const defaultSlots = React.useMemo(() => {
+    const slots = {
+      footer: showBanner ? <Footer /> : null,
+      initial: status === "initial" ? <Initial /> : undefined,
+      authenticating:
+        status === "authenticating" ? (
+          <Authenticating flowState={flowState} />
+        ) : undefined,
+      success:
+        status === "success" ? <Success flowState={flowState} /> : undefined,
+      error: status === "error" ? <Error flowState={flowState} /> : undefined,
+    };
+
+    return slots;
+  }, [status, showBanner, flowState]);
+
+  const slots = useSlots({ children, defaultSlots });
+
+  const handleSubmit = useCallback(
+    (factor: Factor, handle?: Handle) => {
+      if (flowState.status !== "initial") return;
+
+      flowState.logIn(
+        {
+          factor,
+          handle,
+        },
+        { middleware }
+      );
+    },
+    [flowState, middleware]
+  );
 
   return (
-    <div className={clsx("sid-form", styles.form, className)}>
-      <ConfigurationOverrides text={text} factors={factors}>
-        {flowState.status === "initial" && (
-          <FormProvider>
-            <Initial flowState={flowState} lastHandle={lastHandle} middleware={middleware} />
-          </FormProvider>
-        )}
-        {flowState.status === "authenticating" && (
-          <FormProvider>
-            <Authenticating flowState={flowState} />
-          </FormProvider>
-        )}
-        {flowState.status === "error" && <Error flowState={flowState} />}
-        {flowState.status === "success" && <Success flowState={flowState} />}
-        {showBanner ? <Footer /> : null}
-      </ConfigurationOverrides>
-    </div>
+    <InternalFormContext.Provider
+      value={{
+        flowState,
+        lastHandle,
+        handleSubmit,
+        submitPayloadRef,
+        selectedFactor,
+        setSelectedFactor,
+      }}
+    >
+      <div className={clsx("sid-form", styles.form, className)}>
+        <ConfigurationOverrides text={text} factors={factors}>
+          {flowState.status === "initial" && (
+            <FormProvider>{slots.initial}</FormProvider>
+          )}
+          {flowState.status === "authenticating" && (
+            <FormProvider>{slots.authenticating}</FormProvider>
+          )}
+          {flowState.status === "success" && slots.success}
+          {flowState.status === "error" && slots.error}
+          {slots.footer}
```

Comment:
was rendering this regardless for `showBanner` value intentional?

Reply:
It's already handled on top, when constructing the default slots.